automerge-c/README.md

```diff
@@ -0,0 +1,95 @@
+
+## Methods we need to support
```

Comment:
This README is a bit confusing with no context, is this a design document for now?

Reply:
Yes - this is just a design document - this is early stage dev - I was just wanting to make a PR and maybe get it into the main branch to keep the merges simpler (since its isolated and wont break the other projects).  We can also leave it as a PR for a while as we make it more real